SimPEG/Mesh/DiffOperators.py

```diff
@@ -565,7 +565,58 @@ def projNeumannOut(n, bc):
 
         return Pbc, Pin, Pout
 
-
+    
+    def unitCellGradx():
+        doc = """Cell centered Gradient in the x dimension used for 
+                regularization. The gradient operator is square (nC-by-nC)"""
+        def fget(self):
+            if self.dim < 3: return None
+            if getattr(self, '_unitCellGradx', None) is None:
+                
+                n = self.vnC
+                gx = ddx(n[0]-1)
+                gx_square = sp.vstack((gx,gx[-1,:]*-1), format="csr")
+                
+                self._unitCellGradx = kron3(speye(n[2]), speye(n[1]), gx_square)
+        
+            return self._unitCellGradx
+        return locals()
+    unitCellGradx = property(**unitCellGradx())
```

Comment:
This is an older notation for the `@property` wrapper code. It is exactly the same.

`locals` provides a dictionary, which is unpacked into the `property`:

```
unitCellGradx = property(fget=fgetFunction, doc=doc)
```

The original `unitCellGradx` is then overwritten by this property. It is a bit messier and harder to follow, but exactly the same. Also, it is how all the diff ops code is written (from 2+ years ago!).